documentation/modules/creating-storage-mapping.adoc

```diff
@@ -8,6 +8,11 @@
 
 You can create a storage mapping by using the {ocp} web console to map source disk storages to {virt} storage classes.
 
+[NOTE]
+====
+You can specify the source and target providers but leave the specific storage mappings empty for future use if VMs currently have no storages. This kind of mapping is called an _empty mapping_.
```

Comment:
same here - there's nothing special about these empty mappings, they are just a concrete case
in https://github.com/kubev2v/forklift-console-plugin/pull/498 we enabled to specify empty mappings since VMs in RHV that only have LUN disks, have no storage domains to map to storage classes - I'd put a note in the UI and CLI sections that talk about plan creation, that say that storage/network mappings should always be specified when creating a plan even when the migrated VMs have no network/storage

Reply:
actually that note should probably be only in the CLI section since users cannot skip creating/selecting mappings in the UI

Reply:
BTW, choosing this solution was easier for us than totally skipping on the mapping steps in the wizard, but we can consider doing that with the refactored new dialogs if required.

Reply:
> ... but we enhanced that as a feature to say that you can always create a plan with unmapped networks/storage, just to avoid blocking the plan creation, and it can be filled out later on once relevant.

to me this approach seems to go against the recent changes we've made in the UI to improve the user experience - as a user, I'd prefer the UI to block me from posting invalid input rather than posting it and then set a failure on the entity that would block me from going forward. in this case, I'd prefer to cancel the plan creation, create a mapping and then create the plan. btw, does it even work when the mapping is updated? AFAIR the UI clones the template mapping - do you go over the clones as well? @sgratch

Reply:
@sgratch @ahadas It sounds like there are two issues here. First, the phrase "empty mapping" isn't useful, so I can just remove the second sentence in the note. Second, do we want to call attention to this functionality? If not, it makes sense to remove the note altogether. But...maybe we want to leave the first sentence of the note in but add a warning such as "Be sure the mapping is still valid before attempting to implement it"? [Phrasing suggestions are welcome.]

Reply:
@RichardHoch +1 on dropping the "empty mapping" part
about the other part - in general, our documentation should reflect how simple it is to migrate using MTV rather than describing all the features (and in this case, I'm not even sure if this feature makes sense as it raises few questions, e.g., what if I create a plan with "valid" mappings and then someone changes it and make the plan invalid - this is something we should avoid)

Reply:
@ahadas @RichardHoch 

Regarding the current functionality discussion:
> to me this approach seems to go against the recent changes we've made in the UI to improve the user experience - as a user, I'd prefer the UI to block me from posting invalid input rather than posting it and then set a failure on the entity that would block me from going forward. in this case, I'd prefer to cancel the plan creation, create a mapping and then create the plan. 

Correct,  please remember that this plan wizard is a legacy code. For 2.5.0, the current wizard behavior avoids blocking plan creation for invalid/wrong/empty mappings, since the assumption was that we didn't want to block it. So the plan is created with an error status and a detailed tooltip of, e.g:

![Screenshot from 2023-09-11 16-17-51](https://github.com/kubev2v/forklift-documentation/assets/18169498/8e823759-c555-447a-955e-70a31d5d9621)

We'll need to handle that as part of the refactored new dialog/wizard. I'll suggest opening an issue on this.


> btw, does it even work when the mapping is updated? AFAIR the UI clones the template mapping - do you go over the cloned mappings as well? @sgratch

No we don't. It will be left as is.
In case you edit the plan, you might get a warning on the mapping step that the previous mapping is invalid, but it won't be related to the original template and it's not auto updated once the template is updated.